src/utils/geoutils.py

```diff
@@ -14,14 +26,57 @@ def pixel_to_geo_coordinates(
     x, y = pixel_coordinates
     image_w, image_h = input_image_dims
     top_lat, top_long, bottom_lat, bottom_long = input_geo_coordinates
-    if image_w < x < 0 or image_h < y < 0:
-        raise PixelOutOfBoundsException()
+    if image_w < x or x < 0 or image_h < y or y < 0:
+        raise PixelOutOfBoundsException(pixel_coordinates, input_image_dims)
     if top_lat > bottom_lat:
         top_lat, bottom_lat = bottom_lat, top_lat
     if top_long > bottom_long:
         top_long, bottom_long = bottom_long, top_long
     return top_lat + x / image_w * (bottom_lat - top_lat), top_long + x / image_w * (bottom_long - top_long)
 
 
-if __name__ == '__main__':
-    print(pixel_to_geo_coordinates((12, 13), (20, 20), (48.17, 27.19, 50.12, 28.22)))
+def get_coordinates_from_image(image_path_or_fp: Union[str, os.PathLike], *pixels: Union[int, Tuple[int, int]]) -> \
+        List[Tuple[float, float]]:
+    """
+    Transform the provided pixel coordinates into geocoordinates.
+    >>> get_coordinates_from_image(
+    >>>     'image.tiff',
+    >>>     10, 20,
+    >>>     (20, 30)
+    >>> )
+
+    Args:
+        pixels: A set of pixel coordinates.
+        image_path_or_fp: A path to a GeoTIFF image
+
+    Returns:
+        The geolocation of the provided pixel coordinates
+    """
+    import rasterio
```

Comment:
usually import is on top of the file. is it desired to be here? import module only on call